### PR TITLE
retry ethgasstation api if no response returned

### PR DESCRIPTION
**Pull request by akremstudy**
#116

retries to the api request until it gets a response.
file changes in both telliot-feed-examples and telliot-core

no tests added yet. tested manually.

### CODE REVIEW

File: src/telliot_feed_examples/reporters/interval.py [hunk 1]
```diff
@@ -117,6 +117,8 @@ async def ensure_staked(self) -> Tuple[bool, ResponseStatus]:
         status = ResponseStatus()
 
         gas_price_gwei = await self.fetch_gas_price()
+        if gas_price_gwei is None:
+            return None, status
```
Comment (oraclown) [inline src/telliot_feed_examples/reporters/interval.py:121]: Return an error status reporting unable to fetch gas price

Comment (oraclown) [reply src/telliot_feed_examples/reporters/interval.py:121]: Something like this: `None, error_status(note, log=logger.error, e=e)`

File: src/telliot_feed_examples/reporters/interval.py [hunk 2]
```diff
@@ -242,6 +244,10 @@ async def ensure_profitable(
                 gas_price = await ethgasstation(style=self.gas_price_speed)
                 self.legacy_gas_price = gas_price
 
+            if self.legacy_gas_price is None:
+                status.ok = False
+                return status
```
Comment (oraclown) [inline src/telliot_feed_examples/reporters/interval.py:249]: Same about `error_status()`, and shouldn't this one also return a bool?

File: src/telliot_feed_examples/reporters/interval.py [hunk 3]
```diff
@@ -389,6 +395,9 @@ async def report_once(
             # Fetch legacy gas price if not provided by user
             if not self.legacy_gas_price:
                 gas_price = await self.fetch_gas_price(self.gas_price_speed)
+                if gas_price is None:
+                    status.ok = False
+                    return None, status
```
Comment (oraclown) [inline src/telliot_feed_examples/reporters/interval.py:400]: `error_status('msg about unable to get gas price')`

**Review by oraclown — CHANGES_REQUESTED** [review]
Nice! I think it needs a couple edits. Also, a test to confirm your changes work.

File: src/telliot_feed_examples/reporters/interval.py (hunk 2 above)
Comment (akremstudy) [reply src/telliot_feed_examples/reporters/interval.py]: added error_status(). The return type is ResponseStatus in this function.

**Review by oraclown — APPROVED** [review]

**Review by oraclown — APPROVED** [review]

**Pull request merged**